sagemaker/launch/launch_infer.py

```diff
@@ -118,7 +118,6 @@ def run_job(input_args, image, unknownargs):
         role=role,
         instance_count=instance_count,
         instance_type=instance_type,
-        model_uri=model_artifact_s3,
```

Comment:
This is why it downloads the model twice, right? Once by SageMaker and once by download_model(model_artifact_s3, model_path, sagemaker_session).

Can you add the above root cause in PR description?

Reply:
What does the model_uri=model_artifact_s3 here actually download? Does it only download the sagemaker artifact?

Reply:
AFAIK SageMaker will download whatever is under model_artifact_s3. It expects an SM-formatted model file, that is a `model.tar.gz`.

There's some of that logic in the SM SDK: https://github.com/aws/sagemaker-python-sdk/blob/a58654ecbd5997aa59e4fcd8389715fe04c71f5d/src/sagemaker/estimator.py#L1881